tests/solidity_files/IntegrationTestContract.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity >=0.8.0;
+
+/// @notice Contract for integration testing of EVM opcodes.
+/// @author Kakarot9000 
+/// @dev Do add functions and storage variables for opcodes accordingly.
+contract IntegrationTestContract {
+    /*//////////////////////////////////////////////////////////////
+                            METADATA STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    /*//////////////////////////////////////////////////////////////
+                               CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /*//////////////////////////////////////////////////////////////
+                            FUNCTIONS FOR OPCODES
+    //////////////////////////////////////////////////////////////*/
+    function opcodeAddress()
+        public
+        view
+        returns (address selfAddress) 
+    {
+        return (address(this));
+    }
+}
```

Comment:
Entrypoint here. I created this solidity file to the integration test for the opcodes, do you think this is a good idea? Open to discuss

Reply:
I really like this, open for other views but this helps a lot for:
- separation of concerns -> you're not testing the addresses of erc20 or erc721
- this is very visual